iterators.tex

```diff
@@ -408,7 +409,7 @@
 \begin{itemize}
 \item \tcode{++i} is valid if and only if \tcode{i++} is valid.
 \item If \tcode{i} is incrementable, then both \tcode{++i}
-  and \tcode{i++} advance \tcode{i} to the next element.
+  and \tcode{i++} advance \tcode{i} to denote the next (possibly past-the-end) element.
```

Comment:
Do we even *need* `WeaklyIncrementable`, or can we replace all uses with `Iterator`?